File: CS50x - 2024/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int cash;
    int coin = 0;
    do
    {
        cash = get_int("Change owed: ");
    }
    while (cash < 0);

    if (cash >= 25)
        do
        {
            coin++;
            cash -= 25;
        }
        while (cash >= 25);

    if (cash >= 10)
        do
        {
            coin++;
            cash -= 10;
        }
        while (cash >= 10);

    if (cash >= 5)
        do
        {
            coin++;
            cash -= 5;
        }
        while (cash >= 5);

    if (cash >= 1)
        do
        {
            coin++;
            cash -= 1;
        }
        while (cash >= 1);

    printf("%i\n", coin);
}

File: CS50x - 2024/sentimental-cash/cash.py
from cs50 import get_float

coin = 0

while True:
    cash = get_float("Change: ")
    if cash > 0:
        break

while cash >= 0.25:
    cash -= 0.25
    coin += 1
    cash = round(cash, 2)

while cash >= 0.10:
    cash -= 0.10
    coin += 1
    cash = round(cash, 2)

while cash >= 0.05:
    cash -= 0.05
    coin += 1
    cash = round(cash, 2)

while cash >= 0.01:
    cash -= 0.01
    coin += 1
    cash = round(cash, 2)

print(coin)
